test/system_test.py

```diff
@@ -101,52 +117,74 @@ def spack_install_and_test_python_package(command: str,
-    #     spack_installcosmo_and_test(f'cosmo @6.0 %{nvidia_compiler} cosmo_target=gpu +cppdycore ^{mpi} %{nvidia_compiler}')
+    def test_install_version_6_0(self):
+        spack_install_and_test(
+            f'cosmo @6.0 %{nvidia_compiler} cosmo_target=cpu ~cppdycore ^{mpi} %{nvidia_compiler}'
+        )
+        spack_install_and_test(
+            f'cosmo @6.0 %{nvidia_compiler} cosmo_target=gpu +cppdycore ^{mpi} %{nvidia_compiler}'
+        )
 
-    # def test_install_version_5_09_mch_1_2_p2_cpu(self):
-    #     spack_installcosmo_and_test(
-    #         f'cosmo @5.09a.mch1.2.p2 %{nvidia_compiler} cosmo_target=cpu ~cppdycore ^{mpi} %{nvidia_compiler}'
-    #     )
+    @pytest.mark.no_daint  # Patches are not applied. Therefore the tests fail.
+    def test_devbuild_version_6_0_cpu(self):
+        # to avoid cloning into the same folder and having race conditions
+        unique_folder = uuid.uuid4().hex
+
+        subprocess.run(
+            f'git clone --depth 1 --branch 6.0 git@github.com:COSMO-ORG/cosmo.git {unique_folder}',
+            check=True,
+            shell=True)
+        spack_devbuild_and_test(
+            f'cosmo @dev_build_6.0_cpu %{nvidia_compiler} cosmo_target=cpu ~cppdycore ^{mpi} %{nvidia_compiler}',
+            cwd=unique_folder)
+
+    @pytest.mark.no_daint  # Patches are not applied. Therefore the tests fail.
+    def test_devbuild_version_6_0_gpu(self):
+        # to avoid cloning into the same folder and having race conditions
+        unique_folder = uuid.uuid4().hex
+
+        subprocess.run(
+            f'git clone --depth 1 --branch 6.0 git@github.com:COSMO-ORG/cosmo.git {unique_folder}',
+            check=True,
+            shell=True)
+        spack_devbuild_and_test(
+            f'cosmo @dev_build_6.0_gpu %{nvidia_compiler} cosmo_target=gpu +cppdycore ^{mpi} %{nvidia_compiler}',
+            cwd='cosmo')
+
+    @pytest.mark.no_daint  # Testsuite fails
+    def test_install_version_5_09_mch_1_2_p2_cpu(self):
+        spack_install_and_test(
+            f'cosmo @5.09a.mch1.2.p2 %{nvidia_compiler} cosmo_target=cpu ~cppdycore ^{mpi} %{nvidia_compiler}'
+        )
 
-    # def test_install_version_5_09_mch_1_2_p2_gpu(self):
-    #     spack_installcosmo_and_test(
-    #         f'cosmo @5.09a.mch1.2.p2 %{nvidia_compiler} cosmo_target=gpu +cppdycore ^{mpi} %{nvidia_compiler}'
-    #     )
+    @pytest.mark.no_daint  # Unable to open MODULE file gt_gcl_bindings.mod
+    def test_install_version_5_09_mch_1_2_p2_gpu(self):
+        spack_install_and_test(
+            f'cosmo @5.09a.mch1.2.p2 %{nvidia_compiler} cosmo_target=gpu +cppdycore ^{mpi} %{nvidia_compiler}'
+        )
 
     def test_install_c2sm_master_cpu(self):
-        spack_installcosmo_and_test(
+        spack_install_and_test(
             f'cosmo @c2sm-master %{nvidia_compiler} cosmo_target=cpu ~cppdycore ^{mpi} %{nvidia_compiler}'
         )
 
+    @pytest.mark.no_daint  # Unable to open MODULE file gt_gcl_bindings.mod
```

Comment:
What happened here?
This test was green for sure at some point.

Reply:
I don't know. I wasn't able to figure it out. It worked here: https://github.com/C2SM/spack-c2sm/pull/598 , but I can't identify a change that would affect this build.